From 5726fb7b6072b85af4d673bb75c4992c388bef17 Mon Sep 17 00:00:00 2001
From: Eric Timmons <etimmons@mit.edu>
Date: Sat, 14 Apr 2018 21:38:51 -0400
Subject: [PATCH] Test for NPTL explicitly only when using glibc

If we're not using glibc (e.g., musl), then just assume that the thread
implementation is good enough.
---
 src/runtime/linux-os.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/runtime/linux-os.c b/src/runtime/linux-os.c
index fa4f5e490..2c42de016 100644
--- a/src/runtime/linux-os.c
+++ b/src/runtime/linux-os.c
@@ -180,6 +180,9 @@ int linux_sparc_siginfo_bug = 0;
 int
 isnptl (void)
 {
+#ifdef __GLIBC__
+  /* Using GLIBC, so we can test if NPTL is available. Otherwise, just assume we
+   * have a good enough thread implementation. */
   size_t n = confstr (_CS_GNU_LIBPTHREAD_VERSION, NULL, 0);
   if (n > 0) {
       char *buf = alloca (n);
@@ -189,6 +192,9 @@ isnptl (void)
       }
   }
   return 0;
+#else
+  return 1;
+#endif
 }
 #endif
 
-- 
2.17.0

